lib/internal/process/policy.js

```diff
@@ -4,34 +4,78 @@ const {
   JSONParse,
   ObjectFreeze,
   ReflectSetPrototypeOf,
+  SafeMap
 } = primordials;
 
 const {
   ERR_MANIFEST_TDZ,
+  ERR_MANIFEST_ASSERT_INTEGRITY
 } = require('internal/errors').codes;
 const { Manifest } = require('internal/policy/manifest');
+const { pathToFileURL, URL } = require('internal/url');
+const { readFileSync } = require('fs');
+const { isAbsolute } = require('path');
+const { parse } = require('internal/policy/sri');
+
 let manifest;
 let manifestSrc;
 let manifestURL;
 
 module.exports = ObjectFreeze({
   __proto__: null,
-  setup(src, url) {
-    manifestSrc = src;
-    manifestURL = url;
-    if (src === null) {
+  setup(policyPath, policyIntegrity) {
+    // URL here as it is slightly different parsing
+    // no bare specifiers for now
+    let manifestURLObject;
+    if (isAbsolute(policyPath)) {
+      manifestURLObject = new URL(`file://${policyPath}`);
```

Comment:
What if the path contains a char that has a special meaning in URL (e.g. `?`)? @watilde since you're the one who committed this line, do you know if that should be that instead:
```suggestion
      manifestURLObject = pathToFileUrl(policyPath);
```
(non-blocking for this PR)

Reply:
I'd rather leave this to other PRs as this one only tries to move things to a different palce.